data_structures/graphs/prim.c

```diff
@@ -0,0 +1,229 @@
+#include<stdio.h>
+#include<stdlib.h>
+#define max_edges_of_vertex 100000
+/*      max_edges_of_vertex will be used to initialize adjacency list
+        you can modify size according to question constraint                        */
+
+#define INFINITY LONG_LONG_MAX
+
+// a structure to represent a weighted edge in graph
+typedef struct NODE
+{
+	long long u;
+	long long w;
+	struct NODE* next;
+}node;
+
+// a structure to represent a connected, undirected Heap
+typedef struct HeapNode
+{
+	long long u;
+	long long key;
+}heap;
+
+// a function to adjust heap after extracting minimum
+void adjust_min_heap(heap *H, long long i, long long new_weight, long long size, long long *pos) {
```

Comment:
Functions should be documented explaining the use of its parameters, a brief description, and what it returns.